privacyidea/api/healthcheck.py

```diff
@@ -0,0 +1,103 @@
+from flask import Blueprint, current_app
+
+from privacyidea.api.lib.utils import send_result
+from privacyidea.lib.crypto import get_hsm
+from privacyidea.lib.resolver import get_resolver_list, get_resolver_class
+import logging
+
+log = logging.getLogger(__name__)
+
+healthz_blueprint = Blueprint('healthz_blueprint', __name__)
+
+
+@healthz_blueprint.route('/', methods=['GET'])
+def healthz():
+    """
+    Health check endpoint that indicates if the app is healthy (running and ready to serve requests).
+    """
+    app = current_app._get_current_object()
+    if not app.config.get('APP_READY'):
```

Comment:
so currently, when the endpoint is called, the app must already exist. But then the `APP_READY` is always true (unless it set to false again somewhere else).
I guess it is a chicken-egg problem...

Reply:
Correct, the APP_READY flag is currently a placeholder until a more sophisticated method of checking if the server has started is developed. In the future, this flag should be switched to False every time dependencies anywhere in the code are not ready.